sympy/logic/inference.py

```diff
@@ -77,6 +77,12 @@ def satisfiable(expr, algorithm=None, all_models=False, minimal=False):
     UNSAT
 
     """
+    if use_lra_theory:
+        if algorithm is not None:
+            assert algorithm == "dpll2"
```

Comment:
This should use `ValueError` not `AssertionError`. Assertions should not be used to check user inputs.

Reply:
And elsewhere too, an AssertionError should only ever occur if there is a bug in the code.